### PR TITLE
Add v0.4.0 Changelog

### PR DESCRIPTION
**Pull request by hhaAndroid**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation for this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a potential influence on downstream projects, this PR should be tested with downstream projects, like MMDetection or MMClassification.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: docs/en/notes/changelog.md [hunk 1]
```diff
@@ -1,5 +1,29 @@
...
+### Bug Fixes
+
+1. Fix `optimize_anchors.py` script import error (#452)
+2. Fix the wrong installation steps in `get_started.md` (#474)
+
+### Contributors
+
+A total of 9 developers contributed to this release.
+
+Thank VoyagerXvoyagerx, @tianleiSHI, @RangeKing, @PeterH0323, @Nioolek, @triple-Mu, @lyviva, @Zheng-LinXiao, @hhaAndroid
```
Comment (Nioolek) [inline docs/en/notes/changelog.md:25]: Thank VoyagerXvoyagerx?

File: docs/en/notes/changelog.md [hunk 2]
```diff
@@ -1,5 +1,29 @@
 # Changelog
 
+## v0.4.0 (8/1/2023)
+
+### Highlights
+
+1. Implemented [YOLOv8](https://github.com/open-mmlab/mmyolo/blob/dev/configs/yolov8/README.md) object detection model, and supports model deployment for \[projects/easydeploy\](https:// github.com/open-mmlab/mmyolo/blob/dev/projects/easydeploy)
```
Comment (PeterH0323) [inline docs/en/notes/changelog.md]:
```suggestion
1. Implemented [YOLOv8](https://github.com/open-mmlab/mmyolo/blob/dev/configs/yolov8/README.md) object detection model, and supports model deployment for [projects/easydeploy](https:// github.com/open-mmlab/mmyolo/blob/dev/projects/easydeploy)
```

File: docs/en/notes/changelog.md [hunk 3]
```diff
@@ -1,5 +1,29 @@
 # Changelog
 
+## v0.4.0 (18/1/2023)
+
+### Highlights
+
+1. Implemented [YOLOv8](https://github.com/open-mmlab/mmyolo/blob/dev/configs/yolov8/README.md) object detection model, and supports model deployment for [projects/easydeploy](https://github.com/open-mmlab/mmyolo/blob/dev/projects/easydeploy)
```
Comment (PeterH0323) [inline docs/en/notes/changelog.md:7]: 建议将 【训练推理】 和 【部署】 分开写

File: docs/en/notes/changelog.md [hunk 4]
```diff
@@ -1,5 +1,29 @@
...
+2. Added Chinese and English versions of [Algorithm principles and implementation with YOLOv8](https://github.com/open-mmlab/mmyolo/blob/dev/docs/zh_cn/algorithm_descriptions/yolov8_description.md)
```
Comment (Nioolek) [inline docs/en/notes/changelog.md:8]: 这里用dev分支的链接吗？而且改成英文版的更好

File: docs/en/notes/changelog.md (hunk 3 above)
Comment (hhaAndroid) [reply docs/en/notes/changelog.md:7]: 我感觉这句话比较短，合起来写好看点？

Comment (PeterH0323) [reply docs/en/notes/changelog.md:7]: 可以

Comment (PeterH0323) [inline docs/en/notes/changelog.md:7]:
```suggestion
1. Implemented [YOLOv8](https://github.com/open-mmlab/mmyolo/blob/dev/configs/yolov8/README.md) object detection model, and supports model deployment in [projects/easydeploy](https://github.com/open-mmlab/mmyolo/blob/dev/projects/easydeploy)
```

File: docs/en/notes/changelog.md (hunk 4 above)
Comment (hhaAndroid) [reply docs/en/notes/changelog.md:8]: 这里是写错了，应该用 dev 的 en 链接

**Review by PeterH0323 — APPROVED** [review]
LGTM

**Review by Nioolek — APPROVED** [review]
LGTM

**Pull request merged**